File: 2033. Minimum Operations to Make a Uni-Value Grid.cpp
class Solution {
    public:
        int minOperations(vector<vector<int>>& grid, int x) 
        {
            vector<int> nums;
            for(auto &vec : grid)
                for(auto &val : vec) 
                    nums.push_back(val);
            sort(begin(nums), end(nums));
            
            int ans = 0, mid = nums[nums.size() / 2], need;
            for(auto &val : nums)
            {
                need = abs(mid - val);
                if(need % x) return -1;
                ans += need / x;
            }
            return ans;
        }
    };

File: 2033. Minimum Operations to Make a Uni-Value Grid.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        store = []
        for row in grid:
            for val in row:
                store.append(val)
        
        store.sort()
        target = store[len(store) // 2]

        ans = 0
        for val in store:
            extra = abs(val - target)
            if extra % x: return -1
            ans += extra // x
        return ans